app/models/consultation.rb

```diff
@@ -0,0 +1,33 @@
+class Consultation < ApplicationRecord
+	include Paginator
+	
+  belongs_to :ministry
+  belongs_to :created_by, foreign_key: 'created_by_id', class_name: 'User', optional: true
+
+  enum status: [:submitted, :published, :rejected, :expired]
+
+  scope :status_filter, lambda { |status|
+    return all unless status.present?
+    where(status: status)
+  }
+
+  scope :ministry_filter, lambda { |ministry_id|
+    return all unless ministry_id.present?
+    where(ministry_id: ministry_id)
+  }
+
+  def publish
+  	self.status = :published
+  	self.published_at = DateTime.now
+  	self.save!
+  end
+
+  def reject
+  	self.update(status: :rejected)
```

Comment:
Shouldn't we store rejected_at and rejected by?

Reply:
@anbublacky - I kept this simple, I am still waiting for a notification flow from the partner as we also need to add email triggers. Added Issue#1 to track this.